internal/pipe/blob/blob.go

```diff
@@ -0,0 +1,61 @@
+// Package blob provides a Pipe that push artifacts to blob supported by Go CDK
+package blob
+
+import (
+	"fmt"
+
+	"github.com/goreleaser/goreleaser/internal/pipe"
+	"github.com/goreleaser/goreleaser/internal/semerrgroup"
+	"github.com/goreleaser/goreleaser/pkg/context"
+
+	// Import the blob packages we want to be able to open.
+	_ "gocloud.dev/blob/azureblob"
+	_ "gocloud.dev/blob/gcsblob"
+	_ "gocloud.dev/blob/s3blob"
+)
+
+// Pipe for Artifactory
+type Pipe struct{}
+
+// String returns the description of the pipe
+func (Pipe) String() string {
+	return "Blob"
+}
+
+// Default sets the pipe defaults
+func (Pipe) Default(ctx *context.Context) error {
+	for i := range ctx.Config.Blob {
+		blob := &ctx.Config.Blob[i]
+		if blob.Bucket == "" {
+			continue
+		}
+		if blob.Provider == "" {
+			blob.Provider = "azblob"
```

Comment:
I think it's best to leave this as empty by default, and just err earlier `if bucket != "" && provider == ""`

Reply:
yeah, you are right, even setting bucket to "" is not good as Go CDK gonna complain anyway.
`azureblob.OpenBucket: containerName is required` or `s3blob.OpenBucket: bucketName is required`

I would go for something like this then:
```bash
	if blob.Bucket == "" || blob.Provider == "" {
			return fmt.Errorf("bucket or provider cannot be empty")
	}

```
Agree?

Reply:
agree!

Reply:
resolve in 59c8e4